app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java

```diff
@@ -24,6 +55,102 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        eventViewModel = new ViewModelProvider(this).get(EventViewModel.class);
+        tvProfileUsername = view.findViewById(R.id.tvProfileUsername);
+        rvBookmarks = view.findViewById(R.id.rvBookmarks);
+        tvProfileUsername.setText(ParseUser.getCurrentUser().getUsername());
+
+        JsonArray allBookmarks = new JsonArray();
+        setUpRecyclerView(view);
+        getAndSetUserBookmarks(allBookmarks);
+    }
+
+    public void getAndSetUserBookmarks(JsonArray allBookmarks) {
+        bookmarkList.clear();
```

Comment:
❓ Why did you decide to clear the bookmark list first?

Reply:
By clearing the bookmark list, duplicates are prevented from being added to the feed.